lib/puppet/provider/exec/powershell.rb

```diff
@@ -3,13 +3,17 @@
 require File.join(File.dirname(__FILE__), '../../../puppet_x/puppetlabs/powershell/powershell_manager')
 
 Puppet::Type.type(:exec).provide :powershell, :parent => Puppet::Provider::Exec do
-  confine :operatingsystem => :windows
-
-  commands :powershell =>
+  commands :powershell => 
     if File.exists?("#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe")
       "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe"
     elsif File.exists?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
       "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe"
+    elsif File.exists?('/usr/bin/powershell')
+      '/usr/bin/powershell'
+    elsif File.exists?('/usr/local/bin/powershell')
+      '/usr/local/bin/powershell'
+    elsif !Puppet::Util::Platform.windows?
```

Comment:
Do you really expect `powershell` to be somewhere in PATH but not in a `bin` directory?